src/materialized/src/lib.rs

```diff
@@ -289,6 +294,14 @@ pub fn serve(mut config: Config) -> Result<Server, anyhow::Error> {
             executor: &executor,
             experimental_mode: config.experimental_mode,
         })?;
+
+        // Start a thread that checks for the latest version and prints a warning if it
+        // finds a different version than currently running.
+        if let Some(telemetry_url) = config.telemetry_url {
+            let cluster_id = coord.cluster_id().to_string();
+            thread::spawn(move || check_version_loop(telemetry_url, cluster_id));
```

Comment:
We're trying to move away from the blocking APIs. The fact that the coordinator still uses `thread::spawn` is a historical artifact that I've been trying to remove forever. Could you switch this to use `tokio::spawn` and an async function? Should mostly be as simple as using `async fn`s, using the non-blocking reqwest client, and sprinkling in a few `await`s.